src/swarm-engine/Swarm/Game/Tick.hs

```diff
@@ -17,11 +17,14 @@ import Swarm.Util.WindowedCounter (Offsettable (..))
 -- | A newtype representing a count of ticks (typically since the
 --   start of a game).
 newtype TickNumber = TickNumber {getTickNumber :: Int64}
-  deriving (Eq, Ord, Show, Read, Generic, FromJSON)
+  deriving (Eq, Ord, Show, Read, Generic)
 
 instance ToJSON TickNumber where
   toJSON = A.genericToJSON (A.defaultOptions {A.unwrapUnaryRecords = True})
 
+instance FromJSON TickNumber where
+  parseJSON = A.genericParseJSON (A.defaultOptions {A.unwrapUnaryRecords = True})
```

Comment:
Maybe want to define `(A.defaultOptions {A.unwrapUnaryRecords = True})` at the toplevel and share it?

Reply:
Sure, good idea.  Do you think we might want to use it in other modules as well (i.e. should we put it in some kind of util module, or just define it right here)?

Reply:
Util module sounds good.  Perhaps name it `optionsUnwrapUnary`.